docs/full/README.md

```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
 
-`info`: provides metadata about the plugin that the CLI will use when presenting information about plugins or when performing lifecycle operations on them.
+`info`: provides metadata about the plugin that the CLI will use when presenting information about plugins or when
+performing lifecycle operations on them.
 
-`post-install`: provide a means for a plugin to optionally implement some logic to be invoked right after a plugin is installed.
+`post-install`: provide a means for a plugin to optionally implement some logic to be invoked right after a plugin is
+installed.
 
-`generate-docs`: generate a tree of documentation markdown files for the commands the plugin provides, typically used by the CLI's generate-all-docs command to produce command documentation for all installed plugins
+`generate-docs`: generate a tree of documentation markdown files for the commands the plugin provides, typically used by
+the CLI's generate-all-docs command to produce command documentation for all installed plugins
 
 `lint`: validate the command name and arguments to flag any new terms unaccounted for in the CLI taxonomy document
 
-More information about these commands are available in the [plugin contract](../plugindev/contract.md) section of the plugin development guide.
+More information about these commands are available in the [plugin contract](../plugindev/contract.md) section of the
+plugin development guide.
+
+## Secure plugin installation
+
+CLI verifies the identity and integrity of the plugin while installing the plugin
+from the repository. You can find more details in
+the [secure plugin installation proposal document](../design/secure-plugin-installation-design.md)
+
+### User experience
+
+CLI verifies cosign signature of the plugin inventory image present in the
+repository. If the signature verification is success, it would download the
+plugin inventory image on the user's machine and caches to improve the latency
+on subsequent plugin installation/search commands.
+If the signature verification fails, CLI would throw an error and stops continuing.
+Signature verification could fail in the below scenarios
+
+1. Unplanned key rotation: In this case, user either can update to the latest
+   CLI version release with the new key, or users should download the new
+   public key posted in a well known secure location[TBD] to their local file
+   system and export the path of the public key by setting the environment
+   variable `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_PUBLIC_KEY_PATH`.
+2. Repositories without a signature: If users/developers wants to use their own
+   repository without the signature for testing, they can skip the
+   validation(not recommended in production) by appending the repository URL to
+   the comma-separated list in the environment
+   variable `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION_SKIP_LIST`.
```

Comment:
An example, e.g. calling "tanzu config set...." would be nice.

Reply:
Added example to show how user can add 2 plugin repositories to the signature validation skip list.Thanks